osu.Game.Rulesets.Catch.Tests/TestSceneHyperDashColouring.cs

```diff
@@ -0,0 +1,306 @@
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Scale = new Vector2(4f),
+                }, true, false, false);
+            });
+
+            AddStep("start hyper-dashing", () =>
+            {
+                catcherArea.MovableCatcher.SetHyperDashState(2);
+                catcherArea.MovableCatcher.FinishTransforms();
+            });
+
+            AddAssert("catcher use hyper-dash colour from skin", () => catcherArea.MovableCatcher.Colour == TestSkin.CustomHyperDashColour);
+            AddAssert("catcher trails use hyper-dash colour from skin", () => catcherArea.OfType<Container<CatcherTrailSprite>>().Any(c => c.Colour == TestSkin.CustomHyperDashColour));
+
+            AddStep("clear hyper-dash", () =>
+            {
+                catcherArea.MovableCatcher.SetHyperDashState(1);
+                catcherArea.MovableCatcher.FinishTransforms();
+            });
+
+            AddAssert("hyper-dash colour cleared from catcher", () => catcherArea.MovableCatcher.Colour == Color4.White);
+        }
+
+        [Test]
+        public void TestHyperDashCatcherEndGlowColour()
+        {
+            CatcherArea catcherArea = null;
+
+            AddStep("setup catcher", () =>
+            {
+                Child = setupSkinHierarchy(catcherArea = new CatcherArea
+                {
+                    RelativePositionAxes = Axes.None,
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Scale = new Vector2(4f),
+                }, false, false, false);
+            });
+
+            AddStep("start hyper-dashing", () => catcherArea.MovableCatcher.SetHyperDashState(2));
+            AddAssert("end-glow catcher sprite has default hyper-dash colour", () => catcherArea.OfType<Container<CatcherTrailSprite>>().Any(c => c.Colour == Catcher.DefaultHyperDashColour));
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void TestCustomHyperDashCatcherEndGlowColour(bool customHyperDashCatcherColour)
+        {
+            CatcherArea catcherArea = null;
+
+            AddStep("setup catcher", () =>
+            {
+                Child = setupSkinHierarchy(catcherArea = new CatcherArea
+                {
+                    RelativePositionAxes = Axes.None,
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Scale = new Vector2(4f),
+                }, customHyperDashCatcherColour, true, false);
+            });
+
+            AddStep("start hyper-dashing", () => catcherArea.MovableCatcher.SetHyperDashState(2));
+            AddAssert("end-glow catcher sprite use its hyper-dash colour from skin", () => catcherArea.OfType<Container<CatcherTrailSprite>>().Any(c => c.Colour == TestSkin.CustomHyperDashAfterColour));
+        }
+
+        [Test]
+        public void TestCustomHyperDashCatcherEndGlowColourFallback()
```

Comment:
these are too long. remove unnecessary words (the class already asserts most of this, and "custom" is redundant since you already say "fallback")